nx_parallel/utils/chunk.py

```diff
@@ -2,65 +2,94 @@
 import os
 import networkx as nx
 
+
 __all__ = ["chunks", "get_n_jobs", "create_iterables"]
 
 
-def chunks(iterable, n):
-    """Divides an iterable into chunks of size n"""
+def chunks(iterable, n_chunks):
+    """Yield exactly `n_chunks` chunks from `iterable`, balancing the chunk sizes."""
+    iterable = list(iterable)
+    k, m = divmod(len(iterable), n_chunks)
     it = iter(iterable)
-    while True:
-        x = tuple(itertools.islice(it, n))
-        if not x:
-            return
-        yield x
+    for _ in range(n_chunks):
+        chunk_size = k + (1 if m > 0 else 0)
+        m -= 1
+        yield tuple(itertools.islice(it, chunk_size))
 
 
 def get_n_jobs(n_jobs=None):
-    """Returns the positive value of `n_jobs`."""
+    """Returns the positive value of `n_jobs`
+
+    Returns the positive value of `n_jobs` by either extracting it from the
+    active configuration system or modifying the passed-in value, similar to
+    joblib's behavior.
+
+    - If running under pytest, it returns 2 jobs.
+    - If the `active` configuration in NetworkX's config is `True`, `n_jobs`
+      is extracted from the NetworkX config.
+    - Otherwise, `n_jobs` is obtained from joblib's active backend.
+    - `ValueError` is raised if `n_jobs` is 0.
+    """
     if "PYTEST_CURRENT_TEST" in os.environ:
         return 2
-    else:
+
+    if n_jobs is None:
         if nx.config.backends.parallel.active:
             n_jobs = nx.config.backends.parallel.n_jobs
         else:
             from joblib.parallel import get_active_backend
 
             n_jobs = get_active_backend()[1]
-        n_cpus = os.cpu_count()
-        if n_jobs is None:
-            return 1
-        if n_jobs < 0:
-            return n_cpus + n_jobs + 1
-        if n_jobs == 0:
-            raise ValueError("n_jobs == 0 in Parallel has no meaning")
-        return int(n_jobs)
+
+    if n_jobs is None:
+        return 1
+    if n_jobs < 0:
+        return os.cpu_count() + n_jobs + 1
+
+    if n_jobs == 0:
+        raise ValueError("n_jobs == 0 in Parallel has no meaning")
+
+    return int(n_jobs)
 
 
 def create_iterables(G, iterator, n_cores, list_of_iterator=None):
-    """Creates an iterable of function inputs for parallel computation
+    """Create an iterable of function inputs for parallel computation
```

Comment:
More of these minor nit-picky issues:We don't seem to be consistent in what tense the verb leading off the doc_string should use. I think NetworkX has tried to settle on "Create" over "Creates". But there are other doc_strings in this PR (like `get_n_jobs`) which use "Returns".  Rabbit holes abound. :)

Reply:
`create` (i.e. the imperative mood) is indeed the preferred convention 99.99% of the time. The only time I've ever seen the third-person singular form being used is in Java. 

That being said, "Creates" was actually not something that I changed on this PR -- that is what the docstring reads on the `main` branch currently! That being said, happy to change it ;)